### PR TITLE
Csstudio 2331 thread search results (Beta)

### PR DESCRIPTION
**Pull request by cjenkscybercom**
## Summary of Changes

- Updates beta search result list:
  - single items include tags and logbooks as pills/chips
  - group/reply items render in "thread" appearance that sorts in same order as list
  - arrow indicates group member that matches the "parent" that appeared in the search results, so you can see the context of e.g. your search result within a string of replies (chronologically-speaking). Again, bearing in mind that replies aren't heirarchical; they're just a flat list from an association to a group id.
- Adds search title, counts, sort button, and filter button inline above list
  - note the filter button is a stand-in and does nothing at the moment
- Adds relevant Storybook stories

Remember to run with `npm run start-beta` when reviewing these changes, as they are only available in Beta.

Screenshot 2024-04-18 at 16 54 56

Screenshot 2024-04-18 at 16 30 26

## Visual Inspection

- [ ] Conformance to Markdown styles: https://olog.esss.lu.se/Olog/help/CommonmarkCheatsheet
    - [ ] ...when viewing a log entry
    - [ ] ...when previewing HTML while writing a description
    - [ ] ...when viewing a log entry in the group view
- [ ] Scroll is possible (elements don't overflow their container, and they are scrollable)
    - [ ] ...search result list
    - [ ] ...log entry group view list
    - [ ] ...log entry single view
    - [ ] ...create new log entry page
- [ ] Overall layout fills full width and height of viewport
- [ ] Pagination element doesn't overflow into other elements

### CODE REVIEW

**Comment by jacomago** [comment]
Some questions:

1. Are the colours customizable and/or removable? The orange is difficult to read for me. 
2. What does the arrow character that starts the first result mean?
3. Maybe the filter button could be invisible until it works?
4. What is the actual search term in the screenshots? I find it hard to judge what was just searched.

File: src/stories/beta/SearchResultList.stories.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      minHeight: "90vh"
+    }}>
+      <SearchResultList onRowClick={onRowClick} {...props} />
+    </Paper>
+  )
+
+}
+export const Default = (args) => <SearchResultListTemplate {...args} />;
+Default.args = {
+  logs: [log0, log1, log2, log3, log4, log5],
```
Comment (jacomago) [inline src/stories/beta/SearchResultList.stories.js:117]: Personally I would write a generator function to create examples. Then test against different amounts of results:

no results, 1 result, < 10 results, > paging amount of results.

And then same for number of tags and logbooks.

Just an idea. 🔧

**Comment by cjenkscybercom** [comment]
> Some questions:
> 
> 1. Are the colours customizable and/or removable? The orange is difficult to read for me.
> 2. What does the arrow character that starts the first result mean?
> 3. Maybe the filter button could be invisible until it works?
> 4. What is the actual search term in the screenshots? I find it hard to judge what was just searched.

1. They aren't atm, wasn't something considered in the scope of this. Any suggested colors instead? See http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--default and http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--complementary-colors
2. The arrow indicates the log entry that is the parent, e.g. shows the parent in the context of the conversation. 
3. It's easier to refine things that exist, so I want to leave that in there. Some folks just can't picture things that aren't in front of them.
4. Not in the scope of this request. Atm it's searching everything until the BE is updated to allow the "everything" term.

**Review by jacomago — COMMENTED** [review]
Another idea I have is that I think Typography allows you to create a them. Then you can specify a bunch of styles for certain things, like formattedDateStyle. Then use those instead of "body2" with "itatlic", or "body2" with "bold" etc. 

I think this helps in making the hierachy of styles in one place, and clearly separated from the display.

**Comment by jacomago** [comment]
> 1. They aren't atm, wasn't something considered in the scope of this. Any suggested colors instead? See http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--default and http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--complementary-colors

I'm more thinking you have one place you define "border color" or "highlight result color" and then put instead of grey.300.

File: src/stories/beta/SearchResultList.stories.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import { rest } from "msw";
+import { sbDisabledArg } from "stories/sb-utils";
+
+export default {
+  title: "beta/SearchResultList"
+}
+
+let createdDate = moment();
+const replyGroup = group({});
+const log0 = {
```
[the remark this answers is not in the document]
Comment (cjenkscybercom) [reply src/stories/beta/SearchResultList.stories.js:15]: I already am? The ... operator is copying the properties of the previous entry, and then I'm overwriting only the ones I care about.

File: src/beta/components/search/SearchResultList/SearchResultSingleItem.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export const SearchResultSingleItem = ({log, selected, onClick}) => {
+
+  return (
+    <Stack 
+      sx={{
+        "&:hover": {
+          cursor: "pointer",
+          backgroundColor: "grey.300"
```
[the remark this answers is not in the document]
Comment (cjenkscybercom) [reply src/beta/components/search/SearchResultList/SearchResultSingleItem.jsx:41]: I don't want to do this yet, it's too soon to do this when feedback will significantly change this in the next refinement.

File: src/stories/beta/SearchResultList.stories.js (hunk 1 above)
Comment (cjenkscybercom) [reply src/stories/beta/SearchResultList.stories.js:117]: Again, too much effort for one story. I'll pull them out into a utility when I need to, but not sooner.

**Comment by cjenkscybercom** [comment]
> > 1. They aren't atm, wasn't something considered in the scope of this. Any suggested colors instead? See http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--default and http://ccce.pages.esss.lu.se/dev/ce-ui-common/?path=/story/styles-colors--complementary-colors
> 
> I'm more thinking you have one place you define "border color" or "highlight result color" and then put instead of grey.300.

Let's take that as a separate improvement. That's outside the scope of this work, thanks!

**Comment by cjenkscybercom** [comment]
Also: 
Double checked the component tests still actually pass (locally); please ignore the flaky tests here.

**Review by georgweiss — COMMENTED** [review]
I like the search result list UI, it conveys key information in a compact manner.
Some minor remarks:
The count pill overlaps with the reply icon, see screen shot. Maybe we should skip the reply icon?
The "tree" element pushes the attachment icon to the right, which break alignment with other items in the list, see screen shot:

Screenshot 2024-04-25 at 12 36 06

**Comment by cjenkscybercom** [comment]
> I like the search result list UI, it conveys key information in a compact manner. Some minor remarks: The count pill overlaps with the reply icon, see screen shot. Maybe we should skip the reply icon? The "tree" element pushes the attachment icon to the right, which break alignment with other items in the list, see screen shot:
> 
> Screenshot 2024-04-25 at 12 36 06

Thanks! Let's note both those on 

> I like the search result list UI, it conveys key information in a compact manner. Some minor remarks: The count pill overlaps with the reply icon, see screen shot. Maybe we should skip the reply icon? The "tree" element pushes the attachment icon to the right, which break alignment with other items in the list, see screen shot:
> 
> Screenshot 2024-04-25 at 12 36 06

Thanks! I've noted those points on https://jira.esss.lu.se/browse/CSSTUDIO-2374 since they're minor and mostly aesthetic; let's address them there as part of refinement. FTR the overlap of the badge with the icon is intentional, as the default positioning puts it outside the box too much. (also, see https://www.setproduct.com/blog/badge-ui-design as an example where overlap is used in compact designs)

**Comment by georgweiss** [comment]
Still thing the overlap between badge and icon is aesthetically problematic as it is difficult to discern the icon. Since the badge already conveys the information that this is a "thread", the icon is in my view not needed as there is no case where only the icon would be displayed. By definition a thread/group must contain at least two entries.

**Review by georgweiss — APPROVED** [review]

**Pull request merged**